pkg/migration/plan_executor.go

```diff
@@ -103,14 +111,15 @@ func NewPlanExecutor(plan Plan, executors []Executor, opts ...PlanExecutorOption
 
 func (pe *PlanExecutor) Execute() error { //nolint:gocyclo // easier to follow this way
 	ctx := make(map[string]any)
-	for i := 0; i < len(pe.plan.Spec.Steps); i++ {
+	for i := pe.StartIndex; i < len(pe.plan.Spec.Steps); i++ {
 		var r CallbackResult
 		if pe.callback != nil {
 			r = pe.callback.StepToExecute(pe.plan.Spec.Steps[i], i)
 			switch r.Action {
 			case ActionCancel:
 				return nil
 			case ActionSkip:
+				pe.LastSuccessfulStep = i
```

Comment:
How do we use `LastSucccessfulStep`? If a step is skipped, is it still a successful one?

Reply:
I left that up to the reviews to talk about. In family-migrator, we return the skip action for the steps that the user has run manually. We should consider successful steps that are run manually and that the user validates it.

However, the skip action is a much more general action. Therefore, we can specify another type of action for such situations. (For manually performed steps)

Reply:
Or the another option can be that changing the name of the `LastSucccessfulStep` to `LastPerformedStep`.

This field is used for caching mechanism for plan. Please see this [PR](https://github.com/upbound/extensions-migration/pull/18)